### PR TITLE
DPL Analysis: easy way to filter a table on the fly

### PR DESCRIPTION
**Pull request by aalkin**
@jgrosseo @saganatt I've realized there is an easier way to filter a table on-the-fly

### CODE REVIEW

**Comment by saganatt** [comment]
Nice! :-)

File: Framework/Core/include/Framework/ASoA.h [hunk 1]
```diff
@@ -1000,6 +1003,13 @@ class Table
     mBegin.bindExternalIndices(current...);
   }
 
+  auto Select(framework::expressions::Filter&& f) const
```
Comment (ktf) [inline Framework/Core/include/Framework/ASoA.h]: Lowercase...

Comment (aalkin) [reply Framework/Core/include/Framework/ASoA.h]: FIxed!

**Comment by ktf** [comment]
Don't we have `select(tracks, expressions)` already? If not, can we add that as well?

**Comment by aalkin** [comment]
> Don't we have `select(tracks, expressions)` already? If not, can we add that as well?

You mean as an external function? We have
```
template <typename T>
auto filter(T&& t, framework::expressions::Filter const& expr)
```
though it is not used anywhere.

UPD: I fixed the `filter` function to work just like `select`.

**Comment by jgrosseo** [comment]
This is very nice!
Can we move it to a different tutorial though? We are using the full histograms for benchmarking, and I don't like to change it so that we can keep a full evolution of the time needed as a function of release tag.
You can add it to the histograms one instead (or a new one) if you want

File: Framework/Core/include/Framework/ASoA.h [hunk 2]
```diff
@@ -1763,9 +1773,9 @@ template <typename T>
 using is_soa_filtered_t = typename framework::is_base_of_template<soa::FilteredPolicy, T>;
 
 template <typename T>
-auto filter(T&& t, framework::expressions::Filter const& expr)
+auto filter(T const& t, framework::expressions::Filter&& f)
 {
-  return Filtered<T>(t.asArrowTable(), expr);
+  return t.select(f);
```
Comment (ktf) [inline Framework/Core/include/Framework/ASoA.h]: I think the other way around would be better: define the object oriented code in terms of the functional one. Also I would rename it consistently either both select or both filter.

Comment (aalkin) [reply Framework/Core/include/Framework/ASoA.h]: Understood, I will move the functionality to an external function and use other tutorial example.

**Comment by aalkin** [comment]
@jgrosseo what do you think of the new example?

**Comment by jgrosseo** [comment]
I guess this is what we should use as well for the MC associations...

**Pull request merged**